### PR TITLE
chore(lib-inject): allow all users to read and modify installed packages

### PR DESCRIPTION
**Pull request by mabdinur**
## Descriptions

When library injection is used the `/datadog-lib/ddtrace_pkg` directory is created with permissions that prevents other users from reading certain files. This PR ensures all files in`/datadog-lib/ddtrace_pkgs` are readable to all users.

When  `find /datadog-lib/ddtrace_pkgs ! -perm -a=r` is run on a host using library injection with `ddtrace==1.17.3` the following output is generated:

<details>
  /datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.11-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.7-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.7-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.7-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.7-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.7-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.7-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.8-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.8-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.8-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.8-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.8-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.8-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.9-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.9-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/ddtrace_pkgs/site-packages-ddtrace-py3.9-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.10-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.10-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.11-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.7-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.7-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.7-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.7-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.7-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.7-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.8-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.8-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.8-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.8-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.8-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.8-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-manylinux2014/protobuf-4.23.4.dist-info/WHEEL
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-musllinux_1_1/protobuf-4.23.4.dist-info/METADATA
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-musllinux_1_1/protobuf-4.23.4.dist-info/RECORD
/datadog-lib/ddtrace_pkgs/site-packages-ddtrace-py3.9-musllinux_1_1/protobuf-4.23.4.dist-info/WHEEL
</details>

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: lib-injection/Dockerfile [hunk 1]
```diff
@@ -29,6 +29,7 @@ ARG UID=10000
 RUN addgroup -g 10000 -S datadog && \
     adduser -u ${UID} -S datadog -G datadog
 RUN chown -R datadog:datadog /datadog-init/ddtrace_pkgs
+RUN chmod -R a+rwx /datadog-init/ddtrace_pkgs
```
Comment (mabdinur) [inline lib-injection/Dockerfile:32]: a+rwx is very permissive but I think this is okay.

File: .github/workflows/lib-injection.yml [hunk 2]
```diff
@@ -77,6 +77,12 @@ jobs:
           # Wait for the app to start
           sleep 60
           docker-compose logs
+      - name: Check Permissions on ddtrace pkgs
+        run: |
+           # Ensure /datadog-lib/ddtrace_pkgs is a valid directory that contains ddtrace packages
+           docker run -it lib_inject find /datadog-lib/ddtrace_pkgs | wc -l && if [ $? -eq 0 ]; then exit 1; fi
+           # Ensure all users have read permissions to files stored in /datadog-lib/ddtrace_pkgs
+           docker run -it lib_inject find /datadog-lib/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```
Comment (mabdinur) [inline .github/workflows/lib-injection.yml:85]:
```suggestion
           # Ensure /datadog-lib/ddtrace_pkgs is a valid directory that contains ddtrace packages
           docker run -i lib_inject find /datadog-lib/ddtrace_pkgs | wc -l && if [ $? -eq 0 ]; then exit 1; fi
           # Ensure all users have read permissions to files stored in /datadog-lib/ddtrace_pkgs
           docker run -i lib_inject find /datadog-lib/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```

File: .github/workflows/lib-injection.yml [hunk 3]
```diff
@@ -77,6 +77,12 @@ jobs:
           # Wait for the app to start
           sleep 60
           docker-compose logs
+      - name: Check Permissions on ddtrace pkgs
+        run: |
+           # Ensure /datadog-lib/ddtrace_pkgs is a valid directory that contains ddtrace packages
+           docker run -i lib_inject find /datadog-lib/ddtrace_pkgs | wc -l && if [ $? -eq 0 ]; then exit 1; fi
+           # Ensure all users have read permissions to files stored in /datadog-lib/ddtrace_pkgs
+           docker run -i lib_inject find /datadog-lib/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```
Comment (mabdinur) [inline .github/workflows/lib-injection.yml:85]:
```suggestion
      - name: Check Permissions on ddtrace pkgs
        run: |
           # Ensure /datadog-lib/ddtrace_pkgs is a valid directory that contains ddtrace packages
           docker run lib_inject find /datadog-lib/ddtrace_pkgs | wc -l && if [ $? -eq 0 ]; then exit 1; fi
           # Ensure all users have read permissions to files stored in /datadog-lib/ddtrace_pkgs
           docker run lib_inject find /datadog-lib/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```

File: .github/workflows/lib-injection.yml [hunk 4]
```diff
@@ -77,6 +77,13 @@ jobs:
           # Wait for the app to start
           sleep 60
           docker-compose logs
+      - name: Check Permissions on ddtrace pkgs
+        run: |
+           cd lib-injection
+           # Ensure /datadog-lib/ddtrace_pkgs is a valid directory that is not empty
+           docker run lib-injection-lib_inject find /datadog-init/ddtrace_pkgs -maxdepth 0 -empty | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```
Comment (mabdinur) [inline .github/workflows/lib-injection.yml:84]: This test is failing

Comment (Kyle-Verhoog) [inline .github/workflows/lib-injection.yml:84]:
```suggestion
           docker-compose run lib_inject find /datadog-init/ddtrace_pkgs -maxdepth 0 -empty | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```

File: .github/workflows/lib-injection.yml [hunk 5]
```diff
@@ -77,6 +77,13 @@ jobs:
...
+           # Ensure all users have read permissions to files stored in /datadog-lib/ddtrace_pkgs
+           docker run lib-injection-lib_inject find /datadog-init/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```
Comment (mabdinur) [inline .github/workflows/lib-injection.yml:86]:
```suggestion
           docker-compose run lib_inject find /datadog-init/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```

**Comment by stefansedich** [comment]
Ah great this will solve my problem, I had just been working on a PR to datadog-agent to set the runAsUser/runAsGroup on the init container but this will do the trick!

File: releasenotes/notes/lib-injection-pkg-dir-permissions-aaac0059b6c21f30.yaml [hunk 6]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    lib-injection: Resolves permissions errors raised when ddtrace packages are copied from init container to a host's volume.
```
Comment (mabdinur) [inline releasenotes/notes/lib-injection-pkg-dir-permissions-aaac0059b6c21f30.yaml:4]:
```suggestion
    lib-injection: Resolves permissions errors raised when ddtrace packages are copied to a host's volume.
```

File: releasenotes/notes/lib-injection-pkg-dir-permissions-aaac0059b6c21f30.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
+---
+fixes:
+  - |
+    lib-injection: Resolves permissions errors raised when ddtrace packages are copied to a host's volume.
```
Comment (Kyle-Verhoog) [inline releasenotes/notes/lib-injection-pkg-dir-permissions-aaac0059b6c21f30.yaml:4]:
```suggestion
    lib-injection: Resolves permissions errors raised when ddtrace packages are copied from the InitContainer to the shared volume.
```

File: lib-injection/Dockerfile (hunk 1 above)
Comment (Kyle-Verhoog) [reply lib-injection/Dockerfile:32]: do any of the files have the execute permission (prior to us setting it)? If not, then I think we should avoid setting it

**Review by Kyle-Verhoog — COMMENTED** [review]
Looks good to me, just wanna check the execute permission

File: lib-injection/Dockerfile (hunk 1 above)
Comment (mabdinur) [inline lib-injection/Dockerfile:32]:
```suggestion
RUN chmod -R a+rw /datadog-init/ddtrace_pkgs
```

File: .github/workflows/lib-injection.yml [hunk 8]
```diff
@@ -82,6 +82,13 @@ jobs:
           # Wait for the app to start
           sleep 60
           docker-compose logs
+      - name: Check Permissions on ddtrace pkgs
+        run: |
+           cd lib-injection
+           # Ensure /datadog-lib/ddtrace_pkgs is a valid directory that is not empty
+           docker-compose run lib_inject find /datadog-init/ddtrace_pkgs -maxdepth 0 -empty | wc -l && if [ $? -ne 0 ]; then exit 1; fi
+           # Ensure all users have read permissions to files stored in /datadog-lib/ddtrace_pkgs
+           docker-compose run lib_inject find /datadog-init/ddtrace_pkgs ! -perm -a=r | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```
Comment (mabdinur) [inline .github/workflows/lib-injection.yml:91]:
```suggestion
           docker-compose run lib_inject find /datadog-init/ddtrace_pkgs -maxdepth 0 -empty | wc -l && if [ $? -ne 0 ]; then exit 1; fi
           # Ensure non-datadog users have read and write permissions to files stored in /datadog-lib/ddtrace_pkgs
           docker-compose run lib_inject find /datadog-init/ddtrace_pkgs ! -perm -a=rw | wc -l && if [ $? -ne 0 ]; then exit 1; fi
```

**Review by Yun-Kim — APPROVED** [review]

**Review by Kyle-Verhoog — APPROVED** [review]

**Pull request merged**